### DISCUSSION TITLE
Getting Error on Mirth Server Startup

### DISCUSSION THREAD

**Discussion by PrajwalZade** [nextgenhealthcare/connect#5787] [General]
Hi Team,
I'm using Mirth server 4.2.0 32bit.
I experienced random startup error when I did Mirth Restart or Laptop Restart.
Errors are like 
1. ArrayIndexOutOfBoundException
2. NullPointerException

Configurations Using:
1. Java1.8 32bit
2. SSMS SQL server
3. Mirth 4.2.0 32bit
<img width="465" alt="error" src="https://github.com/nextgenhealthcare/connect/assets/59258346/1055bbb1-fb81-4749-ade6-d5c46a3c77d1">

Also log is not generating for this issue. 

Any leads would be helpful. Thanks in advance.

**Comment by jonbartels** [comment]
A few things jump out at me:

* *32bit* unless you have a very good reason to run 32 bit, use 64-bit. The only good reason to use 32-bit would be if you are running Mirth on old hardware or old OS that does not support 64-bit
* Which Mirth application is throwing the error? That looks like the Windows Mirth Manager app that handles config and controls the service. The other relevant apps are the server itself, the client, and the MC Admin Launcher
* This could be a file permissions issue. What user and group owns the `conf` directory for your server installation?

The line shown in the stack trace is https://github.com/nextgenhealthcare/connect/blob/e0db5497aabb94c41071e4a098580e660a1d5d8e/manager/src/com/mirth/connect/manager/ManagerDialog.java#L546 . This line is looking for a `log4j2.properties` file from the Mirth service and the first `rootLogger` element in that file. 

Are you just wanting to run Mirth on a laptop for development?
